src/worker.js

```diff
@@ -6,19 +6,32 @@ const cls = require('continuation-local-storage').createNamespace('ponos')
 const clsBlueBird = require('cls-bluebird')
 const defaults = require('101/defaults')
 const ErrorCat = require('error-cat')
-const exists = require('101/exists')
-const isNumber = require('101/is-number')
 const isObject = require('101/is-object')
+const joi = require('joi')
 const merge = require('101/put')
 const monitor = require('monitor-dog')
-const pick = require('101/pick')
 const Promise = require('bluebird')
 const uuid = require('uuid')
 const WorkerStopError = require('error-cat/errors/worker-stop-error')
 
 const TimeoutError = Promise.TimeoutError
 clsBlueBird(cls)
 
+const optsSchema = joi.object({
+  attempt: joi.number().required(),
+  done: joi.func().required(),
+  errorCat: joi.object(),
+  finalRetryFn: joi.func(),
+  job: joi.object().required(),
+  log: joi.object().required(),
+  maxNumRetries: joi.number(),
+  msTimeout: joi.number().min(0),
```

Comment:
NM. The code was doing something else... Integer would not be a bad idea. Not sure you want a float here.